src/Build/Evaluation/Conditionals/StringExpressionNode.cs

```diff
@@ -42,11 +42,21 @@ internal override bool BoolEvaluate(ConditionEvaluator.IConditionEvaluationState
         /// </summary>
         internal override double NumericEvaluate(ConditionEvaluator.IConditionEvaluationState state)
         {
+            if (ShouldBeTreatedAsVisualStudioVersion())
+            {
+                return ConversionUtilities.ConvertDecimalOrHexToDouble(MSBuildConstants.CurrentVisualStudioVersion);
+            }
+
             return ConversionUtilities.ConvertDecimalOrHexToDouble(GetExpandedValue(state));
         }
 
         internal override Version VersionEvaluate(ConditionEvaluator.IConditionEvaluationState state)
         {
+            if (ShouldBeTreatedAsVisualStudioVersion())
+            {
+                return Version.Parse(MSBuildConstants.CurrentVisualStudioVersion);
```

Comment:
We should have this parsed already or store its parsed value

Reply:
I thought about doing this and decided not to because:

* This is no more expensive than what we would always have been doing before.
* The pre-parsed value would take up memory for little value.